tests/core/test_utilities.py

```diff
@@ -913,6 +914,81 @@ def test_fit_plane_to_points():
     )
 
 
+DEFAULT_PRINCIPAL_AXES = [[0.0, 0.0, 1.0], [0.0, 1.0, 0.0], [-1.0, 0.0, 0.0]]
```

Comment:
Why is `[z, y, -x]` the default? Is this just what `eigh()` does?

Reply:
Yes this is just what it puts out by default. I can update the name to make this clear?

I do have intentions for a future PR to flip / project / reorder the vectors as needed or where possible to make them better line up with the positive XYZ axes by default though. E.g. such that the default output would be the identity matrix in this case.

Reply:
> Yes this is just what it puts out by default. I can update the name to make this clear?

Since this is just a test this is not too critical to me (comment can also work before the definition, but again, not critical).

> I do have intentions for a future PR to flip / project / reorder the vectors as needed or where possible to make them better line up with the positive XYZ axes by default though. E.g. such that the default output would be the identity matrix in this case.

OK, but that will interact with the current API where vectors are sorted by decreasing size, right? You don't have the freedom to change anything beyond changing 2 vector signs, unless there are degenerate axes, but we're talking about floats so telling "degenerate" is a bit messy issue.

Reply:
> OK, but that will interact with the current API where vectors are sorted by decreasing size, right? You don't have the freedom to change anything beyond changing 2 vector signs, unless there are degenerate axes, but we're talking about floats so telling "degenerate" is a bit messy issue.

The order must be decreasing, yes. But for symmetric datasets two eigenvalues may be almost equal (up to some tolerance). So in these cases I don't see why you can't swap/reorder two vectors if they both explain the same amount of variance.

Reply:
Yes, that is "degeneracy". In fact you can use any two perpendicular vectors from that 2d subspace (plane). As I said, choosing when we have degeneracy can be a bit tricky.

Reply:
> Why is `[z, y, -x]` the default? Is this just what `eigh()` does?

I've added a comment to make this a bit more clear.

Further discussion about degeneracy or swapping vectors can be saved for another PR if/when that feature comes up.